docs/proposals/pod-resource-management.md

```diff
@@ -401,6 +402,39 @@ Update Kubelet’s metrics provider to include Pod level metrics. Use cAdvisor's
 
 This feature will be opt-in in v1.4 and an opt-out in v1.5. We recommend users to drain their nodes and opt-in, before switching to v1.5, which will result in a no-op when v1.5 kubelet is rolled out.
 
+## Implementation Status
+
+The implementation goals of the first milestone are outlined below.
+- [x] Finalize and submit Pod Resource Management proposal for the project #26751
+- [x] Refactor qos package to be used globally throughout the codebase #27749 #28093
+- [x] Add interfaces for CgroupManager and CgroupManagerImpl which implements the CgroupManager interface and creates, destroys/updates cgroups using the libcontainer cgroupfs driver. #27755 #28566
+- [x] Inject top level QoS Cgroup creation in the Kubelet and add e2e tests to test that behaviour. #27853
+- [x] Add PodContainerManagerImpl Create and Destroy methods which implements the respective PodContainerManager methods using a cgroupfs driver. #28017
+- [x] Have docker manager create container cgroups under pod level cgroups. Inject creation and deletion of pod cgroups into the pod workers. Add e2e tests to test this behaviour. #29049
+- [x] Add support for updating policy for the pod and QoS containers. Add e2e tests to test this behaviour. #29087
+- [ ] Figure out a method to restart containers having Restart Policy: Never, which are killed intermittently when they are first started.
+- [ ] Manage QoS level cgroup parameter updates when a pod is added or deleted from the kubelet. Can use cadvisor's watch method to track updates/changes on the cgroup mount One option is to do this through a reconcilation loop that ensures pod cgroups on the cgroup mount are in the desired state. ie. only the desired cgroups exist each with the desired resource parameters. This loop would also ensure that the top level cgroups exist, So it will recreate the top level cgroups if they are deleted by some rogue process on the system.
```

Comment:
Another way to state this would be as follows:

kubelet needs to ensure that the cgroup settings are what the kubelet expects them to be. If security is not of concern, one can assume that once kubelet applies cgroups setting successfully, the values will never change unless kubelet changes it. If security is of concern, then kubelet will have to ensure that the cgroup values meet its requirements and then continue to watch for updates to cgroups via inotify and re-apply cgroup values if necessary.
Updating QoS limits needs to happen before pod cgroups values are updated. When pod cgroups are being deleted, QoS limits have to be updated after pod cgroup values have been updated for deletion or pod cgroups have been removed. Given that kubelet doesn't have any checkpoints and updates to QoS and pod cgroups are not atomic, kubelet needs to reconcile cgroups status whenever it restarts to ensure that the cgroups values match kubelet's expectation.